File: backend/src/main/java/com/andrew/hdss/dtos/LocationDto.java
package com.andrew.hdss.dtos;

import com.andrew.hdss.models.Location;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class LocationDto {
    private Long id;
    private String name;
    private String type;
    private Long parentId;
    private String code;

    public static LocationDto from(Location location) {
        return new LocationDto(
                location.getId(),
                location.getName(),
                location.getType().name(),
                location.getParent() != null ? location.getParent().getId() : null,
                location.getCode()
        );
    }
}

File: android/app/src/main/java/com/andrew/hdss/data/dtos/LocationDto.kt
package com.andrew.hdss.data.dtos

import android.util.Log
import com.andrew.hdss.data.models.Location
import com.andrew.hdss.data.models.enums.LocationType
import kotlinx.serialization.Serializable

@Serializable
data class LocationDto(
    val id: Long,
    val name: String,
    val type: String,
    val parentId: Long?,
    val code: String?
)

fun LocationDto.toEntity(): Location? {
    val locationType = try {
        LocationType.valueOf(type)
    } catch (e: IllegalArgumentException) {
        Log.w("LocationMapper", "Unknown LocationType '$type' for location id=$id — skipping")
        return null
    }

    return Location(
        id = id,
        name = name,
        type = locationType,
        parentId = parentId,
        code = code
    )
}